Int8 AvgPool and Reduce test degradation with OpenVINO in 4.x

### System Information

Branch: 4.x
OS: Linux
DNN backend: OpenVINO
CI: https://github.com/opencv/opencv/actions/runs/23056066957/job/66969813332?pr=28636

```
[ RUN      ] Test_Int8_layers.AvePooling/1, where GetParam() = NGRAPH/CPU
unknown file: Failure
C++ exception with description "OpenCV(4.14.0-pre) /home/openvino/opencv/modules/dnn/src/ie_ngraph.cpp:374: error: (-2:Unspecified error) in function 'initPlugin'
> Failed to initialize Inference Engine backend (device = CPU): Check 'false' failed at src/inference/src/core.cpp:114:
> Supported primitive descriptors list is empty for node: AvgPool_1852912
> " thrown in the test body.
[  FAILED  ] Test_Int8_layers.AvePooling/1, where GetParam() = NGRAPH/CPU (11 ms)
[ RUN      ] Test_Int8_layers.MaxPooling/0, where GetParam() = OCV/CPU
[       OK ] Test_Int8_layers.MaxPooling/0 (2 ms)
[ RUN      ] Test_Int8_layers.MaxPooling/1, where GetParam() = NGRAPH/CPU
[       OK ] Test_Int8_layers.MaxPooling/1 (78 ms)
[ RUN      ] Test_Int8_layers.Reduce/0, where GetParam() = OCV/CPU
[       OK ] Test_Int8_layers.Reduce/0 (1 ms)
[ RUN      ] Test_Int8_layers.Reduce/1, where GetParam() = NGRAPH/CPU
unknown file: Failure
C++ exception with description "OpenCV(4.14.0-pre) /home/openvino/opencv/modules/dnn/src/ie_ngraph.cpp:374: error: (-2:Unspecified error) in function 'initPlugin'
> Failed to initialize Inference Engine backend (device = CPU): Check 'false' failed at src/inference/src/core.cpp:114:
> Supported primitive descriptors list is empty for node: AvgPool_1890083
> " thrown in the test body.
[  FAILED  ] Test_Int8_layers.Reduce/1, where GetParam() = NGRAPH/CPU (3 ms)
```

### Detailed description

-

### Steps to reproduce

-

### Issue submission checklist

- [x] I report the issue, it's not a question
- [x] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [x] I updated to the latest OpenCV version and the issue is still there
- [ ] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/dnn/src/int8layers/pooling_layer.cpp
+++ b/modules/dnn/src/int8layers/pooling_layer.cpp
@@ -296,9 +296,16 @@ public:
                         ov::Shape(pads_begin), ov::Shape(pads_end), ov::Shape(kernel_size),
                         rounding_type, pad_type);
         } else if (type == AVE) {
+#if defined(INF_ENGINE_RELEASE) && INF_ENGINE_VER_MAJOR_GE(2025030000)
+            std::vector<size_t> dilations(kernel_size.size(), 1);
+            pool = std::make_shared<ov::op::v16::AvgPool>(input, ov::Strides(strides), ov::Strides(dilations),
+                        ov::Shape(pads_begin), ov::Shape(pads_end), ov::Shape(kernel_size),
+                        !avePoolPaddedArea, rounding_type, pad_type);
+#else
             pool = std::make_shared<ov::op::v1::AvgPool>(input, ov::Strides(strides),
                         ov::Shape(pads_begin), ov::Shape(pads_end), ov::Shape(kernel_size),
                         !avePoolPaddedArea, rounding_type, pad_type);
+#endif
         } else if (type == SUM) {
             ov::Shape inpShape = input.get_shape();
             CV_Assert(inpShape.size() == 2 + kernel_size.size());

--- a/modules/dnn/test/test_int8_layers.cpp
+++ b/modules/dnn/test/test_int8_layers.cpp
@@ -204,7 +204,10 @@ TEST_P(Test_Int8_layers, AvePooling)
     if (backend != DNN_BACKEND_INFERENCE_ENGINE_NGRAPH)
         testLayer("layer_pooling_ave", "Caffe", 0.0021, 0.0075);
     testLayer("ave_pool_same", "TensorFlow", 0.00153, 0.0041);
-    testLayer("average_pooling_1d", "ONNX", 0.002, 0.0048);
+#if defined(INF_ENGINE_RELEASE) && INF_ENGINE_VER_MAJOR_LT(2025030000)
+    if (backend != DNN_BACKEND_INFERENCE_ENGINE_NGRAPH)
+#endif
+        testLayer("average_pooling_1d", "ONNX", 0.002, 0.0048);
     if (backend != DNN_BACKEND_INFERENCE_ENGINE_NGRAPH)
         testLayer("average_pooling", "ONNX", 0.0014, 0.0032);
     testLayer("average_pooling_dynamic_axes", "ONNX", 0.0014, 0.006);

